core/src/util/deprecations.ts

```diff
@@ -256,3 +262,44 @@ export function reportDeprecatedSyncCommandUsage({
     deprecation,
   })
 }
+
+export function makeDefaultConfigChangeMessage({
+  deprecation,
+  includeLink,
+  style,
+}: {
+  deprecation: Deprecation
+  includeLink?: boolean
+  style?: boolean
+}) {
+  const { featureDesc, hint } = getDeprecations(style ? styles.highlight : (s) => `\`${s}\``)[deprecation]
+
+  const lines = [
+    `The default value of ${featureDesc} configuration will be changed in the next major release, Garden 0.14.`,
+  ]
+
+  if (hint) {
+    lines.push(hint)
+  }
+
+  if (includeLink) {
+    let link = `${DOCS_DEPRECATION_GUIDE}#${deprecation}`
+    if (style) {
+      link = styles.link(link)
+    }
+    lines.push(
+      `To make sure your configuration does not break when we release Garden 0.14, please follow the steps at ${link}`
+    )
+  }
+
+  return lines.join("\n")
+}
+
+export function reportDefaultConfigValueChange({ apiVersion, log, deprecation }: DeprecationWarningParams) {
+  if (apiVersion === GardenApiVersion.v2) {
+    return
+  }
+
+  const warnMessage = makeDefaultConfigChangeMessage({ deprecation, includeLink: true, style: true })
+  emitNonRepeatableWarning(log, `\nDEFAULT CONFIGURATION CHANGE WARNING: ${warnMessage}\n`)
```

Comment:
```suggestion
  emitNonRepeatableWarning(log, `\nDEFAULT VALUE WILL CHANGE: ${warnMessage}\n`)
```

Just some thoughts here: This warning message might be a little bit annoying for users, as there is no way to get rid of the warning other than updating `apiVersion` to v2, which we can only really recommend once we have implemented all the changes.
Should we wait with merging this one until we're done with the other changes?

Also, can we maybe detect that jobs will actually be deployed and only then print the warning, as only then this will be a breaking behavioural change (to reduce the number of people who will see this warning in the first place)

Reply:
Makes sense. It looks like the `waitForJobs` is only used by `Job` status handler. So, we can call the `getWaitForJobs()` helper there and print a warning if no explicit value has been specified.

Reply:
Done in 1994924d0715e5e0f0a48d180662ef99f17d2ef9.

Reply:
Ah, super amazing! That change is fantastic 👍